akka-remote/src/main/scala/akka/remote/artery/TaskRunner.scala

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (C) 2016 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka.remote.artery
+
+import java.util.concurrent.TimeUnit
+
+import scala.util.control.NonFatal
+
+import akka.actor.ExtendedActorSystem
+
+import akka.dispatch.AbstractNodeQueue
+import akka.event.Logging
+import akka.stream.stage.AsyncCallback
+import org.agrona.concurrent.BackoffIdleStrategy
+
+object TaskRunner {
+
+  final class Registration(val task: () ⇒ Boolean) {
+    private[TaskRunner] var active = false
+  }
+
+  sealed trait Command
+  case object Shutdown extends Command
+  final case class Register(
+    registrationCallback: AsyncCallback[Registration],
+    task: () ⇒ Boolean) extends Command
+  final case class Deregister(registration: Registration) extends Command
+  final case class Activate(registration: Registration) extends Command
+
+  final class CommandQueue extends AbstractNodeQueue[Command]
+}
+
+class TaskRunner(system: ExtendedActorSystem) extends Runnable {
+  import TaskRunner._
+
+  private val log = Logging(system, getClass)
+  private[this] var running = false
+  private[this] val cmdQueue = new CommandQueue
+  private[this] val registrations = new java.util.ArrayList[Registration]()
+
+  // TODO the backoff strategy should be measured and tuned
+  private val spinning = 2000000
+  private val yielding = 0
```

Comment:
the samples, which I used for the baseline results are using [busy spin](https://github.com/real-logic/Agrona/blob/master/src/main/java/org/agrona/concurrent/BusySpinIdleStrategy.java), https://github.com/real-logic/Aeron/blob/master/aeron-samples/src/main/java/io/aeron/samples/SamplesUtil.java#L45